packages/ai-jsx/src/batteries/docs.tsx

```diff
@@ -1,154 +1,498 @@
+import { Embeddings } from 'langchain/embeddings/base';
+import { OpenAIEmbeddings } from 'langchain/embeddings/openai';
+import { TokenTextSplitter } from 'langchain/text_splitter';
+import _ from 'lodash';
+import { similarity } from 'ml-distance';
+import pino from 'pino';
+import { Jsonifiable } from 'type-fest';
+import { ChatCompletion, SystemMessage, UserMessage } from '../core/completion.js';
 import * as LLMx from '../index.js';
 import { Node } from '../index.js';
-import { ChatCompletion, SystemMessage, UserMessage } from '../core/completion.js';
-import { Jsonifiable } from 'type-fest';
-import { TokenTextSplitter } from 'langchain/text_splitter';
-import { MemoryVectorStore } from 'langchain/vectorstores/memory';
-import { Document as LangChainDocument } from 'langchain/document';
-import { Embeddings } from 'langchain/embeddings/base';
+
+// DO_NOT_SUBMIT
+const pinoLogger = _.once(() =>
+  // @ts-expect-error
+  pino(
+    { name: 'ai-jsx-docs', level: 'trace' },
+    // N.B. pino.destination is not available in the browser
+    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+    pino.destination?.({
+      dest: './ai-jsx-docs.log',
+      sync: true, // Synchronous logging
+    })
+  )
+)();
```

Comment:
@petersalas, I have [these docs](https://ai-jsx-docs-m4s36woh3-fixie-ai.vercel.app/guides/observability#logging-from-within-components) for how to log within a component, but what are we thinking for this context, where we're not immediately within a component? Maybe this is where we test our "can everything be a component" question? It seems a little clumsy for components to have to pass the logger into every helper they call.

Reply:
(resolving dupe)